src/v2/components/ChannelContents/components/ChannelContentsThumbnail/index.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+
+import {
+  ChannelContents_initial_contents,
+  ChannelContents_initial_contents_Image,
+  ChannelContents_initial_contents_Link,
+  ChannelContents_initial_contents_Embed,
+  ChannelContents_initial_contents_Attachment,
+} from '__generated__/ChannelContents'
+import Image from 'v2/components/UI/Head/components/Image'
+
+interface Props {
+  initialContents: ChannelContents_initial_contents[]
+}
+type ChannelContentsThumbnail =
+  | ChannelContents_initial_contents_Image
+  | ChannelContents_initial_contents_Link
+  | ChannelContents_initial_contents_Embed
+  | ChannelContents_initial_contents_Attachment
+
+const isChannelContentsThumbnail = (
+  connectable: ChannelContents_initial_contents
+): connectable is ChannelContentsThumbnail => {
+  return (connectable as ChannelContentsThumbnail).meta_image !== undefined
+}
```

Comment:
@broskoski PSA: type guards in JSX, took me a sec since the official docs use the JSX-like syntax: https://www.typescriptlang.org/docs/handbook/advanced-types.html vs `as`